### PR TITLE
RT-561: Metrics.IncrementGauge Broken

### PR DESCRIPTION
**Pull request by strvmarv**

### CODE REVIEW

File: Src/StackifyLib/Internal/Metrics/MetricClient.cs [hunk 1]
```diff
@@ -160,39 +168,64 @@ public static LatestAggregate GetLatestMetric(string category, string metricName
...
+
             try
             {
                 //set a sanity cap
-                if (_MetricQueue.Count < 100000)
+                if (MetricQueue.Count < 100000)
+                {
+                    MetricQueue.Enqueue(metric);
+
+                    // RT-561: Incremented Metrics Broken
```
Comment (strvmarv) [inline Src/StackifyLib/Internal/Metrics/MetricClient.cs]: The magic

File: Src/StackifyLib/Internal/Metrics/MetricClient.cs [hunk 2]
```diff
@@ -12,75 +12,76 @@ namespace StackifyLib.Internal.Metrics
...
                     switch (item.Value.MetricType)
                     {
                         case MetricType.Counter:
-                            setting.AutoReportLastValueIfNothingReported = false;//do not allow this
+                            setting.AutoReportLastValueIfNothingReported = false; // do not allow this
                             break;
                         case MetricType.CounterTime:
-                            setting.AutoReportLastValueIfNothingReported = false; //do not allow this
-                            break;
-                        case MetricType.MetricAverage:
```
Comment (eric-martin) [inline Src/StackifyLib/Internal/Metrics/MetricClient.cs]: Why are you removing these types from the switch?

Comment (strvmarv) [reply Src/StackifyLib/Internal/Metrics/MetricClient.cs]: The bottom two have no implementation...might as well fall-through...

Comment (strvmarv) [reply Src/StackifyLib/Internal/Metrics/MetricClient.cs]: I only removed the useless cases...check the side-by-side diff...

Comment (eric-martin) [reply Src/StackifyLib/Internal/Metrics/MetricClient.cs]: Got it.

**Pull request merged**